auth-service/src/test/kotlin/uk/gov/communities/delta/controllers/EditAccessGroupsControllerTest.kt

```diff
@@ -0,0 +1,361 @@
+                    "datamart-delta-access-group-3-orgCode2",
+                    any(),
+                    null
+                )
+            }
+            coVerify(exactly = 1) {
+                groupService.removeUserFromGroup(
+                    externalUser.cn,
+                    externalUser.dn,
+                    "datamart-delta-access-group-1-orgCode3",
+                    any(),
+                    null
+                )
+            }
+            confirmVerified(groupService)
+        }
+    }
+    @Test
+    fun testAccessGroupsForUnselectedOrgsAreNotRemoved() = testSuspend {
+        testClient.post("/bearer/access-groups") {
+            headers {
+                append("Authorization", "Bearer ${externalUserSession.authToken}")
+                append("Delta-Client", "${client.clientId}:${client.clientSecret}")
+            }
+            contentType(ContentType.Application.Json)
+            setBody(
+                "{" +
+                        "\"accessGroupsRequest\": {\"datamart-delta-access-group-1\": [\"orgCode1\", \"orgCode2\"], \"datamart-delta-access-group-2\": [], \"datamart-delta-access-group-3\": []}" +
+                        ", \"userSelectedOrgs\": [\"orgCode1\", \"orgCode2\"]" +
+                        "}"
+            )
+        }.apply {
+            assertEquals(HttpStatusCode.OK, status)
+            coVerify(exactly = 1) {
+                groupService.addUserToGroup(
+                    externalUser.cn,
+                    externalUser.dn,
+                    "datamart-delta-access-group-1-orgCode2",
+                    any(),
+                    null
+                )
+            }
+            coVerify(exactly = 1) {
+                groupService.removeUserFromGroup(
+                    externalUser.cn,
+                    externalUser.dn,
+                    "datamart-delta-access-group-2",
+                    any(),
+                    null
+                )
+            }
+            coVerify(exactly = 1) {
+                groupService.removeUserFromGroup(
+                    externalUser.cn,
+                    externalUser.dn,
+                    "datamart-delta-access-group-2-orgCode1",
+                    any(),
+                    null
+                )
+            }
+            coVerify(exactly = 1) {
+                groupService.removeUserFromGroup(
+                    externalUser.cn,
+                    externalUser.dn,
+                    "datamart-delta-access-group-3",
+                    any(),
+                    null
+                )
+            }
+            coVerify(exactly = 1) {
+                groupService.removeUserFromGroup(
+                    externalUser.cn,
+                    externalUser.dn,
+                    "datamart-delta-access-group-3-orgCode2",
+                    any(),
+                    null
+                )
+            }
+            confirmVerified(groupService)
+        }
+    }
```

Comment:
We should also test that the access group itself isn't removed if the user is a member in another organisation, and that access groups themselves are added if the user isn't already a member. Probably better as a unit tests of just that bit though